algorithmic-strategies/divide-and-conquer.md

```diff
@@ -0,0 +1,292 @@
+
+## Terms
+
+| Term               | Definition                                                                                                                                                                                                                                                                                                             | How to Use in a Sentence                                                                                                   |
+| ------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------- |
+| Divide and Conquer | An algorithm design strategy based on multi-branched recursion. We recursively break down a problem into two or more subproblems of the same or related type, until these become simple enough to be solved directly. The solutions to the subproblems are then combined to give a solution to the original problem. | "If multiple subproblems can be combined to solve a larger problem, we might try to apply a divide-and-conquer solution." |
+| Pivot              | An element used in QuickSort to divide the array into two sections, one section less than the pivot and the other greater than the pivot. The choice of the pivot has enormous implications for the efficiency of QuickSort.                                                                                            | "We should be careful about how we pick the pivot so that we avoid worst-case performance for our typical inputs."                                  |
+
+## Example: Binary Search
+
+Binary search can be considered a divide-and-conquer algorithm.
+
+Starting with a sorted array, we check whether it contains a particular value by comparing the search value with the element in the middle of the array.
+
+If we find the value we return the position where it was found.
+
+If we don't find the value, we determine whether it would be in the left or right half of the array by seeing whether it is smaller or larger than the middle element. Remember, we can make this decision because _we assume the array is sorted!_ Then we perform a binary search on the selected half.
+
+If at any point, we end up with an empty range, we know the value was not in the array, and we can return a result indicating the value was not found, such as `None`. Other variations of binary search may return the index of where the value _should_ have been, but as a negative value to indicate that it was missing.
+
+Each recursion divides the array in half and performs the binary search on a smaller subproblem.
+
```

Comment:
I'd love just a clear sentence that calls out the divide and conquer part. There's a similar one for the QuickSort section ("In terms of divide and conquer, we pick a pivot and move smaller elements ...")

I'm not great the grammarz.

```suggestion

In terms of divide and conquer, this algorithm divides the subarray in half with each step. This algorithm will always reduce the subarrays, and thus the amount of remaining data to search through.

```

Reply:
I see your point, but I think this would be better to go right up at the top when we say binary search can be considered a divide-and-conquer algorithm. Why? this stuff.